Expensify/App
# Add modal for inviting users to a workspace

cc @mountiny 

### Details
This PR adds the ability to invite a user to a workspace.

### Fixed Issues

Fixes https://github.com/Expensify/Expensify/issues/166040

### Tests
0. Go to expensify.com.dev (or expensify.com if testing on prod) and get the policyID of your testing account. You can find this by going to https://www.expensify.com.dev/admin_policies and clicking on one of the policies and you'll see it in the URL. For example, the one I'm using to test is `4A19DC10B34DC505`.
1. Since the workspace people list page isn't done yet, the easiest way to remove users while testing is to go to the URL mentioned above, choose a policy, choose `People` and remove that way.
2. Update [InitialPage.js](https://github.com/Expensify/Expensify.cash/blob/bb4d1364116c45f0f0f35804b6d3ff9e2df1e161/src/pages/settings/InitialPage.js#L87) and add the following object to menuItems, this will allow us to easily navigate to the modal on all platforms while testing (replace the policyID with your policyID)
    1. `{translationKey: 'common.invite', icon: Profile, action: () => { Navigation.navigate(ROUTES.getWorkspaceInviteRoute('4A19DC10B34DC505')); },},`
3. Run the app and go to settings, and click on the bottom option `Invite` (the one we added in step 2), and confirm that the modal opens with the `Invite` button disabled. Confirm that the welcomeNote placeholder value contains the policy name.
4. Open the `Application` tab of the Chrome inspector (web is the easiest way to test this) and type in `policy_4A19DC10B34DC505` in the filter bar (replace `4A19DC10B34DC505` with your policyID), confirm you see the policy object w/ the employeeList
![image](https://user-images.githubusercontent.com/3981102/121986464-5f4c2880-cd4b-11eb-8b85-3d2a2c496cc5.png)
5. Add a new user via the invite modal, confirm that after hitting the Invite button the modal disappears.
6. Refresh the value in the `Application` tab of the Chrome inspector, confirm that the new email is added
7. In the Chrome inspector, filter using `personalDetails`, confirm the invited user's personal details show up
8. To test the failure case, you can take one of two routes:
  1. Internal to Expensify - update the dev code of the API to throw an Exception inside the `Policy_Employees_Merge` command of the PHP api
  2. External - update the `.then` handler of `Policy_Employees_Merge` in `Policy.js` so that the `data.jsonCode === 200` check is now `data.jsonCode !== 200`
9. Invite another user via the modal, confirm that the modal closes and you see an error message
10. Check the Application tab of the Chrome inspector, confirm that the invited email doesn't show up in the list

### QA Steps
N/A, this will be more easily testable by QA in https://github.com/Expensify/Expensify/issues/166468

### Tested On


### Screenshots


#### Web
<img src='http://g.recordit.co/XH3xTpaL2I.gif' width='400'/>

#### Mobile Web
<img src='https://user-images.githubusercontent.com/3981102/121989571-329b0f80-cd51-11eb-8c4c-88dc441142ec.png' width='400'/>

#### Desktop
<img src='https://user-images.githubusercontent.com/3981102/121989726-74c45100-cd51-11eb-8f1f-3f2b3727e1bd.png' width='400'/>

#### iOS
<img src='https://user-images.githubusercontent.com/3981102/121990637-f10b6400-cd52-11eb-924b-db8983565d4b.png' width='400'/>

#### Android
<img src='https://user-images.githubusercontent.com/3981102/121990855-5a8b7280-cd53-11eb-9daf-314a5b5ac995.png' width='400'/>
